### ISSUE TITLE
Linux: Crash when two databases use hardware key and one is closed without key present

### ISSUE THREAD

**Issue by florealcab** [keepassxreboot/keepassxc#11450]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )

Close a database (if 2 databases opened) cause bugs:
- everytime the link between keepassxc and the browser extension (in chrome or firefox) is lost for the still opened databases: icon show an error, saying unable to connect to keepassxc, if we click to reload it's restored
- sometimes, keepassxc completely crash (close, segment fault or something else?) and we have to restart it

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Open a database
2. Open another database
3. Close the last opened database

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )

KeepassXC is still alive, the still opened database is still connected in keepassxc browser

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )

Sometimes, KeepassXC crashes. If not, the link with keepassxc browser is always lost.

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.7.9
Revision: 8f6dd13

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux (debian 12)
Desktop Env: KDE
Windowing System: X11

**Comment by droidmonkey** [comment]
I have never seen the application crash when closing a database tab. You'll have to provide a stack trace to validate that finding. If your other database is not connected to the browser extension, then it will not shift the connection to it. I have also not experienced this issue. When closing a database or changing tabs, the browser extension updates immediately.

**Comment by droidmonkey** [comment]
Please provide a screen recording showing what you are seeing.

**Comment by varjolintu** [comment]
If you are using two databases at the same time, make sure the second one is not a clone. Goto Database Settings -> Browser Integration and generate new Root Group UUID's to them just in case.

**Comment by florealcab** [comment]
I can't give you a video of the crash yet as it's random.

But for the keepassxc browser link broken after closing one of the 2 databases, here is the video:

https://github.com/user-attachments/assets/0f71c3f3-ee2f-44e9-9243-3bd1cad78752

Note that databases are NOT a clone, they have been created the two from scratch (one of them created from a passman CSV export, the other as a new empty db)

**Comment by varjolintu** [comment]
What does the extension log says? You can access it from about:debugging.

**Comment by florealcab** [comment]
The message says (it's in french sorry): [Error keepass.js:256] KeePassXC-Browser - 5: Impossible de se connecter à KeePassXC. Vérifiez que l’intégration au navigateur est activée dans les paramètres de KeePassXC.

The browser integration is activated. If I click to reload button ("Recharger" in french, I think it should be reload in english, or refresh) the connection work again on the remaining database.

**Comment by varjolintu** [comment]
This is something I could try to reproduce, but I'm more interested about the KeePassXC crash.

**Comment by florealcab** [comment]
I will try to reproduce it at my work office soon. It happen more often here than at home for some reason I don't know... and I will try to have some logs about that

**Comment by florealcab** [comment]
I have the bug now in video of keepassxc crash while closing one of the 2 databases:

https://github.com/user-attachments/assets/38347143-fe41-4186-9d57-3c53d8e5a775

I launch keepass from terminal so I can have log. App logs are:

```
...
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
Failed to read proxy message:  "données invalides à la fin du document"
fish: Job 1, 'keepassxc' terminated by signal SIGSEGV (Erreur de frontière d'adresse)
```

First lines appears before crash, not important. The important lines are the 2 last: maybe the crash is due to the french locale so? Maybe the accent in that message?

**Comment by varjolintu** [comment]
Can you provide any crash logs?

**Comment by florealcab** [comment]
I don't know where I find it... KDE don't detect crash of keepass so it does not give me a bug report so... I send you what I have in the terminal from where I launch keepass, I don't have more thant that. Except if you can help me for where I can find it

**Comment by droidmonkey** [comment]
Run a snapshot version to get crash reports: https://snapshot.keepassxc.org

**Comment by florealcab** [comment]
I try the AppImage for the link you give me, but I don't view any crash log too. Dolphin just says me "the process crash" but without more details...

But I better understand how to reproduce the crash error. In my case the work database is protected only by a yubikey. The home database is protected by a yubikey + a password.

Several scenarios:
1. If I close home or work database **with** yubikey still plugged: no crash
2. If I close work database **without** yubikey plugged: no crash
3. If I close home database **without** yubikey plugged: crash always

At home, the yubikey is generally always plugged, but not at work, that seems to explain why I experience this crash bug mainly at work.

Please keep in mind that the bug with keepassxc browser is not related to that, it occurres in all cases listed above.

**Comment by droidmonkey** [comment]
I cannot replicate this crash at all on Windows

**Comment by varjolintu** [comment]
And I cannot replicate the error in the extension when opening two databases and removing the second database tab.

**Comment by droidmonkey** [comment]
I replicated the crash on Linux. I am going to re-title this issue to focus on the crash. If the browser issue persists and we can replicate we can open another issue.

**Comment by droidmonkey** [comment]
I added the fix to #11513
